### DISCUSSION TITLE
Update Chart with Real-Time Tick Data?

### DISCUSSION THREAD

**Discussion by digitalml** [klinecharts/KLineChart#295] [Q&A]
How do we update the chart with real-time tick data without redrawing the entire chart? 

I only have a new datetime, new price and tick volume. (I do not have o,h,l). 

Also, what happens when a new tick with a different minute time comes in?  How do we create that new bar?

There was a discussion on this in https://github.com/liihuu/KLineChart/issues/165 and a PR to add the functionality (doesn't seem the PR is valid any more) 

@liihuu suggested it should not be part of the core library itself and should instead be added manually.  

Can someone please show an example of how we can add real-time tick data manually?

updateData() only accepts all of the bar data (date, o, h, l, c, v) at once and trying to implement it seems confusing. Like I have to add reverse the arrays or something to get it to update?  I can't get it to work at all.

Thank you.

**Comment by mercior** [comment]
Personally I have real-time data working using updateData. I generate candle data on my server as trades come in (in the ohlc format) and pass the current candle across a websocket to the frontend which redirects it into updateData. Works great,

**Comment by digitalml** [comment]
updateData replaces the entire bar, so are you saying that you manually
manage the full time frame candle on the server using each ticks data and
then pass that full candle down to the client each tick?

**Reply by mercior** [reply]
updateData looks at the timestamp of the data you provide it. So if you provide the same timestamp as before, it updates the candle - else, if it sees a new timestamp, it advances the chart and creates a new candle.

So, yes, I manage the candles on the server side. Each trade that comes in I update a candle object. If the timestamp on the trade is past that candles max time, I create a new candle. Then I send the most recent candle data across to the front-end.

**Comment by liihuu** [comment]
Right. Need to format the correct timestamp according to the period and then stuff it into the chart.
